2018-12-26/KBitsCombination.java

```diff
@@ -32,15 +29,20 @@ public static void main(String[] args) {
    * @param n length of the array
    * @param k number of 1 bits to combine
    */
-  private void findAllCombinations(int n, int k) {
+  private static void findAllCombinations(int n, int k) {
     int[] array = new int[n];
     // The array is created with the indices we want to combine
     // if n is 8, an array of [0 1 2 3 4 5 6 7] is created.
-    for(int i=0; i<n; i++) {
+    for (int i = 0; i < n; i++) {
       array[i] = i;
     }
     
-    findAllCombinations(array, new int[k], k, 0, 0);
+    ArrayList<int[]> combinations = new ArrayList<>();
+    KBitsCombination.findAllCombinations(array, new int[k], combinations, k, 0, 0);
+
+    for (int[] c : combinations) {
+      System.out.println(java.util.Arrays.toString(c));
```

Comment:
Awesome. Looks great and thanks! Only remark is the `findAllCombinations` is still really `findAndPrintAllCombinations()` as far as the caller is concerned. You'll give the caller maximum reusability by just returning an `ArrayList<int[]>` and letting them handle the print calls (if they choose).

Reply:
Yeah you're right I missed that one. I'll change it